engines/router/missionctl/errors/errors.go

```diff
@@ -71,49 +73,62 @@ func GetType(err error) ErrorType {
 	return Unknown
 }
 
-// GetHTTPErrorCode maps the ErrorType to http status codes and returns it
-func GetHTTPErrorCode(err error) int {
+// GetErrorCode maps the ErrorType to http status codes and returns it
+func GetErrorCode(err error, p protocol.Protocol) int {
 	var code int
 
+	// Get ErrorType if its turingError else set to default
 	et := GetType(err)
-
-	switch et {
-	case BadInput:
-		code = http.StatusBadRequest
-	case BadResponse:
-		code = http.StatusBadGateway
-	case NotFound:
-		code = http.StatusNotFound
-	default:
-		code = http.StatusInternalServerError
+	if p == protocol.HTTP {
+		switch et {
+		case BadInput:
+			code = http.StatusBadRequest
+		case BadResponse:
+			code = http.StatusBadGateway
+		case NotFound:
+			code = http.StatusNotFound
+		default:
+			code = http.StatusInternalServerError
+		}
+	} else if p == protocol.GRPC {
```

Comment:
There are 2 definition of protocol currently, one in "github.com/gojek/fiber/protocol" another one in "engines/router/missionctl/config/". Can we merge it?

Reply:
my bit of dilemma here is that `fiber` should only have `http` and `grpc` since it is message agnostic. 

In the event of upcoming custom grpc or other message types, Turing would need new `protocol` to identify them, whereas fiber doesn't need to (it will still be a grpc dispatcher regardless). Hence it was intentional to have `UPI` and `HTTP` here.

Reply:
Ah ok, didn't realized the first one is protocol in fiber.

Reply:
@leonlnj Maybe add alias in the import to differentiate between those 2. E.g. 
```
import (

`fiberProtocol "github.com/gojek/fiber/protocol"`

)
```